### ISSUE TITLE
ff-searchbutton: click handler is too restrictive

### ISSUE THREAD

**Issue by fritzmg** [FACT-Finder-Web-Components/ff-web-components#26]
The click handler for the `<ff-searchbutton>` element is too restrictive. It fails to work when the `<button>` element also has children - even if it's just a pseudo child (`::before` or `::after`), because the click handler does the following:

```js
function(e) {
  e.srcElement === this._button && g.a.communication.FFCommunicationEventAggregator.addFFEvent({
    type: "search",
    query: g.a.communication.globalElementValues.currentFFSearchBoxValue || g.a.communication.globalSearchParameter.query
  })
}
```

So if you have something like this:

```html
<ff-searchbutton>
  <button></button>
</ff-searchbutton>
``` 
```css
button::before {
  content: 'Search';
}
```

and you click on `Search`, the search will not be triggered, bcause `e.srcElement === this._button` will be false. Likewise if you have something like

```html
<ff-searchbutton>
  <button><span>Search</span></button>
</ff-searchbutton>
``` 

and you click on `Search`, the search will not be triggered, bcause `e.srcElement === this._button` will be false.

As a workaround you can set

```css
ff-searchbutton button::before,
ff-searchbutton button * {
  pointer-events: none;
}
```

but imho the click handler should be optimized.

**Comment by omikron-fp** [comment]
Thank you for the issue. I have already queued it to be taken care of.

**Comment by omikron-fp** [comment]
Hello @fritzmg,
I couldn't reproduce the behaviour described for the following setup. I am using WebComponents `3.7.0`.
```html
<ff-searchbutton>
  <button></button>
</ff-searchbutton>
```
```css
button::before {
  content: 'Search';
}
```
Clicking on _Search_, `srcElement` seems to always be `button`. Even in IE.

Clicking _Search_ in the following setup will indeed not work out of the box.
```html
<ff-searchbutton>
  <button><span>Search</span></button>
</ff-searchbutton>
```
In this case you have to tell `ff-searchbutton` which element to listen to. You can do so by calling `.resetButton("span")` on `ff-searchbutton` itself. [The documentation](https://web-components.fact-finder.de/api/3.x/ff-searchbox#tab=api) has a small entry for this method.
A possible implementation is to add the following script to your page.
```html
<script>
    window.addEventListener("WebComponentsReady", function() {
        document.querySelector("ff-searchbutton").resetButton("span");
    });
</script>
```
However, I noticed that in IE, `srcElement` will always be `button` even if you click on one of its child elements. So if you must support IE, the only way to consistently make this work at the moment is probably to use a plain `button` element without further nested elements.

That being said, we will keep looking into this issue and try to find a suitable solution.
Changing the click handling behaviour would most likely be breaking, therefore this issue will require careful thought.

**Comment by omikron-fp** [comment]
Fix is currently under review.

**Comment by omikron-fp** [comment]
Fixed with version [3.8.0](https://github.com/FACT-Finder-Web-Components/ff-web-components/releases/tag/3.8.0)
